Documentation/operations/system_requirements.rst

```diff
@@ -171,6 +171,7 @@ linked, either choice is valid.
         CONFIG_CRYPTO_USER_API_HASH=y
         CONFIG_CGROUPS=y
         CONFIG_CGROUP_BPF=y
+        CONFIG_PERF_EVENTS=y
```

Comment:
I'm surprised this isn't listed in https://github.com/cilium/packer-ci-build/blob/master/provision/ubuntu/kernel-next.sh :thinking:

Reply:
Yes, it is required for the BPF stacktrace map support. AFAIK tetragon used it before (and might use it in future).

Reply:
For what it's worth, we had a user report on ARM64 that their kernel did not have this enabled, hence why I raised this PR.

Reply:
Was it causing any issue in Cilium?

Reply:
AFAIK Cilium doesn't use `bpf_ringbuf_*` or stacktrace map. I don't see the immediate dependency on `PERF_EVENTS` for `BPF_MAP_TYPE_PERF_EVENT_ARRAY` but maybe the map can exist and the only way to inject elements into the map is to use other kernel logic that relies on `PERF_EVENTS`?

Reply:
> Was it causing any issue in Cilium?

The user reported the following message before enabling this config option in their kernel:
```
level=fatal msg="Cannot initialise BPF perf ring buffer sockets" error="failed to create perf ring for CPU 0: can't create perf event: function not implemented" startTime="2023-02-27 15:57:02.163467342 +0000 UTC m=+30.470975003" subsys=monitor-agent
```

Then this message appears to have gone away after enabling it.

Reply:
(I'll add the above to the commit message)

Reply:
Ok, this makes sense, the `function not implemented` error is `ENOSYS`. And the ring buffer is initialized using `perf_event_open(2)` which is only available when `CONFIG_PERF_EVENTS` is set (see [kernel/Makefile](https://github.com/torvalds/linux/blob/v6.2/kernel/Makefile#L117)).